app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CategoriesController < ApplicationController
+  load_and_authorize_resource
+  before_action :set_category, only: %i[show]
+
+  def index
+    @categories = current_user.categories
+  end
+
+  def show
+    @deals = @category.deals.order(created_at: 'desc')
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = current_user.categories.new(category_params)
+
+    respond_to do |format|
+      if @category.save
+        format.html { redirect_to root_path, notice: 'Category was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @category.destroy
+    respond_to do |format|
+      format.html { redirect_to root_path, notice: 'Category was successfully deleted.' }
+    end
+  end
```

Comment:
- [ ] Good job adding a notice message when a category is deleted successfully, kindly add a notice message for when a category is not deleted successfully. This will show an error message telling the user that the category was not deleted successfully and improve the user experience.

Reply:
This issue is well resolved! ✔️